content/end-markers-for-method-blocks.md

```diff
@@ -0,0 +1,226 @@
+---
+layout: sip
+permalink: /sips/:title.html
+stage: design
+status: proposal
+presip-thread: https://contributors.scala-lang.org/t/end-markers-for-fewer-braces-blocks/6358
+title: SIP-77 - End Markers for Method Blocks
+---
+
+**By: Oron Port (and AI agent)**
+
+## History
+
+| Date          | Version            |
+|---------------|--------------------|
+| Aug 26th 2025 | Initial Draft      |
+| Oct 24th 2025 | Removed anonymous classes end markers which already exist |
+
+## Summary
+
+This SIP proposes to generalize end markers in Scala 3's indentation-based syntax to cover end markers for blocks that are arguments to method applications, written as `end <methodName>`.
+
+The goal is to improve readability of long braceless blocks, align with existing end markers for definitions, and enable formatter/tooling support without forcing braces. The end marker is optional and may be inserted by tools (e.g., formatters) once a block exceeds a configured length, without changing semantics.
+
+## Motivation
+
+With braceless syntax, large blocks passed as by-name or lambda arguments can be harder to visually pair with their starts, particularly inside nested DSLs, tests, or direct-style code. Contributors have requested optional end markers for such blocks to mirror end markers for definitions and to improve readability and navigation, especially in long, nested structures.
+
+Prior discussions highlight the need to:
+
+- Allow an end marker that refers to the invoked method, e.g., `end test`, `end locally`, which communicates the intent and closes the application site rather than a definition. See the discussion and viewpoints in `End markers for fewer braces blocks` and follow-on threads [link1] and `More indentation syntax end marker variations` [link2].
+
+Illustrative example:
+
+```scala
+def foo(arg: Int)(block: => Unit): Unit = ???
+
+foo(42):
+  // do something
+  // more nested blocks...
```

Comment:
I think all the examples we give (except for some counterexamples) should be valid scala code (currently or after the SIP would become a part of the language). This snippet below is not one, because fewer-braces syntax requires non-empy code blocks (and comments don't count as code in this sence).
The same applies for some other examples below